File: suyeon/Boj10845.java
package suyeon;

import java.io.*;
import java.util.LinkedList;
import java.util.Queue;

public class Boj10845 {
     public static void main(String args[]) throws IOException, Exception {
          BufferedReader br =  new BufferedReader(new InputStreamReader(System.in));
          StringBuilder sb = new StringBuilder();
          Queue<Integer> que = new LinkedList<Integer>();

          //입력받을 테스트 수
          int testCaseCount = Integer.parseInt(br.readLine());
          //큐에 담을 숫자
          int num = 0;

          while (testCaseCount --> 0) {

               //공백으로 구분해준다.
               String[] command = br.readLine().split(" ");
               //String cmd = br.readLine();


               switch (command[0]){
                    case "push":
                         //num = Integer.parseInt(br.readLine());
                         num = Integer.parseInt(command[1]);
                         que.add(num);
                         break;

                    case "pop":
                         sb.append(que.isEmpty()?-1:que.poll()).append("\n");
                         break;

                    case "size":
                         sb.append(que.size()).append("\n");
                         break;

                    case "empty":
                         sb.append(que.isEmpty()?1:0).append("\n");
                         //System.out.println(que.toString());
                         break;

                    case "front":
                         sb.append(que.isEmpty()?-1:que.peek()).append("\n");
                         break;

                    case "back":
                         sb.append(que.isEmpty()?-1:num).append("\n");
                         break;

               }
          }
          System.out.println(sb);
     }
}

File: duna/Boj10845.py
import sys

input=sys.stdin.readline

n=int(input())
que =[] #stack 사용

for _ in range(n):
  command=input().strip()

  if command == 'pop':
    if que:
      print(que.pop(0)) #맨앞의 원소를 뺀것
    else:
      print(-1)

  elif command == 'size':
    print(len(que))
    
  elif command == 'empty':
    if que:
      print(0)
    else:
      print(1)

  elif command == 'front':
    if que:
      print(que[0])
    else:
      print(-1)

  elif command == 'back':
    if que:
      print(que[-1])
    else:
      print(-1)

  else:
    c,value =command.split()
    que.append(value)